indexer/services/roundtable/__tests__/tasks/uncross-orderbook.test.ts

```diff
@@ -0,0 +1,235 @@
+      {},
+      [],
+    );
+    const market: PerpetualMarketFromDatabase = perpetualMarkets[0];
+    await updatePriceLevelWithSleep({
+      ticker: market.ticker,
+      side: OrderSide.BUY,
+      humanPrice: '30100',
+      sizeDeltaInQuantums: '2000',
+      client: redisClient,
+    });
+    await updatePriceLevelWithSleep({
+      ticker: market.ticker,
+      side: OrderSide.SELL,
+      humanPrice: '45000',
+      sizeDeltaInQuantums: '1000',
+      client: redisClient,
+    });
+
+    await runTask();
+    expect(require('@dydxprotocol-indexer/redis/build/src/caches/orderbook-levels-cache').deleteStalePriceLevel).not.toHaveBeenCalled();
+  });
+
+  it('removes single crossed bid level', async () => {
+    const perpetualMarkets: PerpetualMarketFromDatabase[] = await PerpetualMarketTable.findAll(
+      {},
+      [],
+    );
+    const market: PerpetualMarketFromDatabase = perpetualMarkets[0];
+    await OrderbookLevelsCache.updatePriceLevel({
+      ticker: market.ticker,
+      side: OrderSide.BUY,
+      humanPrice: '45100',
+      sizeDeltaInQuantums: '2000',
+      client: redisClient,
+    });
+    await OrderbookLevelsCache.updatePriceLevel({
+      ticker: market.ticker,
+      side: OrderSide.SELL,
+      humanPrice: '45000',
+      sizeDeltaInQuantums: '1000',
+      client: redisClient,
+    });
+
+    await runTask();
+
+    expect(require('@dydxprotocol-indexer/redis/build/src/caches/orderbook-levels-cache').deleteStalePriceLevel).toHaveBeenCalledWith(expect.objectContaining({
+      side: OrderSide.BUY,
+      humanPrice: '45100',
+    }));
+    expect(require('@dydxprotocol-indexer/redis/build/src/caches/orderbook-levels-cache').deleteStalePriceLevel).not.toHaveBeenCalledWith(expect.objectContaining({
+      side: OrderSide.SELL,
+      humanPrice: '45000',
+    }));
+  });
+
+  it('removes first updated when there exist multiple crossed bid levels', async () => {
+    const perpetualMarkets: PerpetualMarketFromDatabase[] = await PerpetualMarketTable.findAll(
+      {},
+      [],
+    );
+    const market: PerpetualMarketFromDatabase = perpetualMarkets[0];
+    await updatePriceLevelWithSleep({
+      ticker: market.ticker,
+      side: OrderSide.BUY,
+      humanPrice: '45100',
+      sizeDeltaInQuantums: '2000',
+      client: redisClient,
+    });
+    await updatePriceLevelWithSleep({
+      ticker: market.ticker,
+      side: OrderSide.SELL,
+      humanPrice: '45000',
+      sizeDeltaInQuantums: '1000',
+      client: redisClient,
+    });
+    await updatePriceLevelWithSleep({
+      ticker: market.ticker,
+      side: OrderSide.BUY,
+      humanPrice: '45200',
+      sizeDeltaInQuantums: '2000',
+      client: redisClient,
+    });
+
+    await runTask();
+
+    expect(require('@dydxprotocol-indexer/redis/build/src/caches/orderbook-levels-cache').deleteStalePriceLevel).toHaveBeenCalledWith(
+      expect.objectContaining({
+        side: OrderSide.SELL,
+        humanPrice: '45000',
+      }));
+  });
```

Comment:
Isn't the BUY at 45100 the first updated? How come that one doesn't get removed?

Reply:
The first crossing buy at price level 45200 is updated last. It's the lowest ask timestamp compared to the highest bid timestamp.

Reply:
Maybe add a comment to note the ordering of how bids / asks are evaluated and removed? Seems like this isn't clear at first glance.

Reply:
done